mllib/src/test/scala/org/apache/spark/mllib/clustering/KMeansSuite.scala

```diff
@@ -85,9 +101,50 @@ class KMeansSuite extends SparkFunSuite with MLlibTestSparkContext {
         Vectors.dense(1.0, 3.0, 4.0)),
       2)
 
-    // Make sure code runs.
-    var model = KMeans.train(data, k = 3, maxIterations = 1)
-    assert(model.clusterCenters.size === 3)
+    var model = KMeans.train(data, k = 3, maxIterations = 1, initializationMode = "random")
+    assert(model.clusterCenters.length === 2)
+
+    model = KMeans.train(data, k = 3, maxIterations = 1, initializationMode = "k-means||")
+    assert(model.clusterCenters.length === 2)
+  }
+
+  test("unique cluster centers") {
+    val rng = new scala.util.Random(42)
+    val points = (0 until 10).map(i => Vectors.dense(Array.fill(3)(rng.nextDouble)))
+    val data = sc.parallelize(
+      points.flatMap { point =>
+        Array.fill(rng.nextInt(4))(point)
+      }, 2
+    )
+    val norms = data.map(Vectors.norm(_, 2.0))
+    val zippedData = data.zip(norms).map { case (v, norm) =>
+      new VectorWithNorm(v, norm)
+    }
+    // less centers than k
+    val km = new KMeans().setK(50)
+      .setMaxIterations(10)
+      .setInitializationMode("k-means||")
+      .setInitializationSteps(10)
+      .setSeed(42)
+    val initialCenters = km.initKMeansParallel(zippedData).map(_.vector)
+    assert(initialCenters.length === initialCenters.distinct.length)
```

Comment:
also check `initialCenters.length <= numDistinctPoints` where `numDistinctPoints` is 10 (defined above) at the moment.